core/arch/arm/kernel/thread_spmc.c

```diff
@@ -322,36 +359,52 @@ static bool is_my_uuid(uint32_t w0, uint32_t w1, uint32_t w2, uint32_t w3)
 	 * The test below is for the hard coded UUID
 	 * 486178e0-e7f8-11e3-bc5e-0002a5d5c51b
 	 */
-	return w0 == 0xe0786148 && w1 == 0xe311f8e7 &&
-	       w2 == 0x02005ebc && w3 == 0x1bc5d5a5;
+	return w0 == my_uuid_words[0] && w1 == my_uuid_words[1] &&
+	       w2 == my_uuid_words[2] && w3 == my_uuid_words[3];
 }
 
-void spmc_fill_partition_entry(struct ffa_partition_info *fpi,
-			       uint16_t endpoint_id, uint16_t execution_context)
+void spmc_fill_partition_entry(uint32_t ffa_vers,
+			       struct ffa_partition_info_x *fpi,
+			       uint16_t endpoint_id,
+			       uint16_t execution_context, uint32_t part_props,
+			       const uint32_t uuid_words[4])
 {
 	fpi->id = endpoint_id;
 	/* Number of execution contexts implemented by this partition */
 	fpi->execution_context = execution_context;
 
-	fpi->partition_properties = FFA_PARTITION_DIRECT_REQ_RECV_SUPPORT |
-				    FFA_PARTITION_DIRECT_REQ_SEND_SUPPORT;
+	fpi->partition_properties = part_props;
+
+	if (ffa_vers >= FFA_VERSION_1_1) {
+		if (uuid_words)
+			memcpy(fpi->uuid, uuid_words, FFA_UUID_SIZE);
+		else
+			memset(fpi->uuid, 0, FFA_UUID_SIZE);
+	}
 }
 
 static uint32_t handle_partition_info_get_all(size_t *elem_count,
-					      struct ffa_rxtx *rxtx)
+					      struct ffa_rxtx *rxtx,
+					      bool count_only)
 {
-	struct ffa_partition_info *fpi = rxtx->tx;
-
-	/* Add OP-TEE SP */
-	spmc_fill_partition_entry(fpi, my_endpoint_id, CFG_TEE_CORE_NB_CORE);
-	rxtx->tx_is_mine = false;
+	struct ffa_partition_info_x *fpi = NULL;
+
+	if (!count_only) {
+		fpi = rxtx->tx;
+		/* Add OP-TEE SP */
+		spmc_fill_partition_entry(rxtx->ffa_vers, fpi, my_endpoint_id,
+					  CFG_TEE_CORE_NB_CORE, my_part_props,
+					  my_uuid_words);
+		rxtx->tx_is_mine = false;
+		fpi++;
```

Comment:
This doesn't take into account that there might be `FFA_UUID_SIZE` number of extra bytes at the end of the `struct ffa_partition_info_x`. I think the increment size should depend on the FF-A version, like in `sp_partition_info_get()`. Probably it'd be easier to move the increment size calculation to a common place, e.g. `spmc_fill_partition_entry()` since there we already have different scenarios based on the FF-A version?

Reply:
Good catch, I'll fix it.